### PR TITLE
C++: Lift `getParameter` to `ParameterNode`

### PR DESCRIPTION
**Pull request by MathiasVP**
cc @pwntester.

### CODE REVIEW

File: cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/DataFlowUtil.qll [hunk 1]
```diff
@@ -1354,6 +1354,9 @@ class ParameterNode extends Node {
    * pointer-indirection parameters are at further negative positions.
    */
   predicate isParameterOf(Function f, ParameterPosition pos) { none() } // overridden by subclasses
+
+  /** Gets the `Parameter` associated with this node, if it exists. */
+  Parameter getParameter() { none() }
```
Comment (MathiasVP) [inline cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/DataFlowUtil.qll:1359]:
```suggestion
  Parameter getParameter() { none() } // overridden by subclasses
```

**Comment by aschackmull** [comment]
Is `Node.asParameter()` not sufficient?

**Comment by MathiasVP** [comment]
> Is `Node.asParameter()` not sufficient?

That would have also worked, yes.

**Review by geoffw0 — APPROVED** [review]
:+1:

**Comment by pwntester** [comment]
Thanks!

**Pull request merged**